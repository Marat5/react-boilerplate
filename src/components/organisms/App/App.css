h1 {
    color: #165e5a
}
