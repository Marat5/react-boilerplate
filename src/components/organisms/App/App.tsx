import './App.css';
import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';

const App: FC = () => {
  const { t, i18n } = useTranslation();

  const changeLanguage = () => {
    i18n.language === 'ru'
      ? i18n.changeLanguage('en')
      : i18n.changeLanguage('ru');
  };
  return (
    <div>
      <h1>{t('template')}</h1>
      <button onClick={changeLanguage}>{t('change_language')}</button>
    </div>
  );
};
export default App;
