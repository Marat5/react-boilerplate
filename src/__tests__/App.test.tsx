import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/organisms/App/App';
import { I18nextProvider } from 'react-i18next';
import i18n from '../i18nForTests';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

it('hello jest', () => {
  const tree = renderer
    .create(
      <I18nextProvider i18n={i18n}>
        <App />
      </I18nextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders a message', () => {
  const { container, getByText } = render(<App />);
  expect(getByText('change language')).toBeInTheDocument();
  expect(container.firstChild?.firstChild?.firstChild).toMatchInlineSnapshot(
    `template`
  );
});
