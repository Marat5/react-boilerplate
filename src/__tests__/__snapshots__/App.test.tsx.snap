// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello jest 1`] = `
<div>
  <h1>
    template
  </h1>
  <button
    onClick={[Function]}
  >
    change language
  </button>
</div>
`;
